### PR TITLE
OIDC Client

### PR DESCRIPTION
**Pull request by Rachana21Reddy**
-OIDC Client implementation
@venusripada @MurthyAvanithsa @sairam1318

### CODE REVIEW

File: src/components/OidcClient.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        const { name, value } = e.target;
+        setValues({
+          ...values,
+          [name]: value,
+        });
+      };
+    const redirect = () => {
+        if((values.authorizeUrl === null) || (values.redirectUrl === null) || (values.clientId === null) || (values.scope === null) || (selectedCheckBox === null) ||(values.responseMode === null)) {
+            if(values.authorizeUrl === null) {
+                document.getElementById("authorizeUrlHelp").innerHTML = "This field is required";
```
Comment (venusripada) [inline src/components/OidcClient.js]: @Rachana21Reddy , this code block needs cleanup

Comment (venusripada) [reply src/components/OidcClient.js]: can we write validation at the element level on blur as well? and call the validation function again when submitting

File: src/components/OidcClient.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            }
+            if(selectedCheckBox === null) {
+                document.getElementById("responseTypeHelp").innerHTML = "This field is required";
+            }  
+            if(values.responseMode === null) {
+                document.getElementById("responseModeHelp").innerHTML = "This field is required";
+            }  
+        }
+        else {
+            let url = values.authorizeUrl+"?client_id="+values.clientId+"&redirect_uri="+values.redirectUrl+"&scope="+values.scope+"&response_type="+selectedCheckBox+"&response_mode"+values.responseMode+"&nonce="+values.nonce+"&state="+values.state;
```
Comment (venusripada) [inline src/components/OidcClient.js]: Validate the authorize url to have URL format with https etc..

**Comment by MurthyAvanithsa** [comment]
Duplicate

**Pull request closed without merging**